### ISSUE TITLE
Function returns T instead of () inspection

### ISSUE THREAD

**Issue by miedzinski** [intellij-rust/intellij-rust#848]
Consider rather simple function

```
fn fun() -> () {
    2 + 2;
}
```

`2 + 2;` is highlighted and intellij-rust suggests an action to remove semicolon, which produces code that does not compile because of mismatched types. Plugin doesn't see this, but there is inspection for that.

Even more weird, with such function (return value omitted)

```
fn fun() {
    2 + 2;
}
```

plugin behaves well, but after manually removing semicolon, the inspection does not trigger.

Overall, seems like this inspection does not work at all. Tested with PyCharm 2016.3.1 and intellij-rust 0.1.0.1286. I guess it broke on https://intellij-rust.github.io/2016/11/14/changelog-18.html; previously I've worked on https://intellij-rust.github.io/2016/10/31/changelog-17.html and it was OK.

**Comment by matklad** [comment]
>but after manually removing semicolon, the inspection does not trigger.

Yeah, we don't have the "reverse" inspection at all. Thanks for the report!
